### PR TITLE
 [SDL 0200] - Removing url parameter max length

### LINKED ISSUE #906: [SDL 0200] - Removing URL Parameter Max Length
Proposal: [Removing URL Parameter Max Length](https://github.com/smartdevicelink/sdl_evolution/blob/master/proposals/0200-Removing-URL-Param-Max-Length.md)

> This proposal is to remove maximum length attribute for URL strings provided to OnSystemRequest, StartStream and StartAudioStream and let the URL strings be of any length permissible by the system architecture.

Review: https://github.com/smartdevicelink/sdl_evolution/issues/603

Steering Committee Decision:

> The Steering Committee accepted this proposal on 10/16/2018. The following was discussed in the Steering Committee meeting on 2018-10-16, with regards to this proposal: Although there are no iOS or Android limitations over the size and length of URLs, excessively long ones may take a long time to create / parse, and extensive data should be sent via a method that is not a URL (e.g. JSON in the bulk data).
Lastly, while it's understood that SDLC members cannot share all use cases for features publicly, it was acknowledged that in the future, when requesting specific features, it is helpful to have real world examples to better understand the reasoning and planned implementation for the requested changes.

### PR DESCRIPTION
**Pull request by kostyaBoss**
Fixes #906 

This PR is **[not ready]** for review.

### Risk
This PR makes **[minor]** API changes.

### Summary
This PR doesn't provide any library changes. Still, it gives unit tests that cover OnSystemRequest class itself with additional checking of the ```url``` parameter to be sure it is not changed in the library over the set/get process.

#### Enhancements

- New unit tests added for OnSystemRequest class

### CLA
- [x] I have signed [the CLA](https://docs.google.com/forms/d/e/1FAIpQLSdsgJY33VByaX482zHzi-xUm49JNnmuJOyAM6uegPQ2LXYVfA/viewform)

### CODE REVIEW

**Comment by kostyaBoss** [comment]
@santhanamk, @mrapitis, @atiwari9 please review changes

**Comment by santhanamk** [comment]
Hi @theresalech.  This PR is ready for Livio review.

**Comment by theresalech** [comment]
@kostyaBoss can you please update the PR description to reflect that this PR is ready for review?  Thanks!

File: android/sdl_android/src/androidTest/java/com/smartdevicelink/test/rpc/notifications/OnSystemRequestTests.java [hunk 1]
```diff
@@ -138,4 +138,27 @@ public void testRpcValues () {
         assertNull(TestValues.NULL, msg.getRequestType());
         assertNull(TestValues.NULL, msg.getRequestSubType());
     }
+
```
Comment (bilal-alsharifi) [inline android/sdl_android/src/androidTest/java/com/smartdevicelink/test/rpc/notifications/OnSystemRequestTests.java:141]: The javadoc for the `OnSystemRequest` still has max length defined. Can you remove that?

Comment (kostyaBoss) [reply android/sdl_android/src/androidTest/java/com/smartdevicelink/test/rpc/notifications/OnSystemRequestTests.java:141]: @bilal-alsharifi Hi! Thanks for your feedback, I've provided changes. Please, check the PR

**Pull request merged**